utils.py

```diff
@@ -60,23 +71,77 @@ def load_cached_dataset(cfg: DictConfig, split="test") -> BinsToVelocityDataset:
     config_hash.update(config_string.encode())
     config_hash = config_hash.hexdigest()
     cache_dir = "tmp/datasets"
+    print(f"Preparing dataset: {config_hash}")
+    try:
+        dataset_cache_file = f"{config_hash}.pkl"
+        dataset_cache_path = os.path.join(cache_dir, dataset_cache_file)
+
+        if os.path.exists(dataset_cache_path):
+            file = open(dataset_cache_path, "rb")
+            dataset = pickle.load(file)
+
+        else:
+            file = open(dataset_cache_path, "wb")
+            hf_dataset = load_dataset(cfg.dataset_name, split=split)
+
+            args = [hf_dataset, n_dstart_bins, n_velocity_bins, n_duration_bins, cfg.sequence_size]
+            try:
+                if cfg.dataset_class == "BinsToDstartDataset":
+                    args.append(cfg.n_tgt_dstart_bins)
+            finally:
+                pass
+
+            dataset = eval(cfg.dataset_class)(*args)
```

Comment:
```suggestion
```

This logic feels too convoluted and too deep into the dataset loading process. I think the decision between different dataset classes should be the first thing to happen:


```python
def load_cached_dataset(
        cfg: DictConfig,
        split: str = "test",
) -> TokenizedMidiDataset:
    if cfg.prediction_target == "velocity":
        dataset = load_cached_velocity_dataset(cfg=cfg, split=split)
    if cfg.prediction_target == "dstart":
        dataset = load_cached_dstart_dataset(cfg=cfg, split=split)
    
    return dataset
```

Reply:
sure, will fix that later